### PR TITLE
Fix cluster creation

### PR DESCRIPTION
**Pull request by korthout**
Adds a workaround to the missing region in the zeebe address url.

It splits the client creation from the cluster creation. This means the cluster is not deleted everytime something goes wrong in the client creation part. 

It also adds a short delay between the client creation and the connection info retrieval, so that console has time to be updated.

The workaround can be removed when camunda-cloud/console#2225 is resolved

### CODE REVIEW

**Comment by Zelldon** [comment]
I see new exceptions @korthout in testbench 

```
java.lang.NullPointerException: RESTEASY004690: Parameter annotated with PathParam cannot be null at java.base/java.util.Objects.requireNonNull(Unknown Source) at org.jboss.resteasy.client.jaxrs.internal.proxy.processors.webtarget.PathParamProcessor.build(PathParamProcessor.java:38) at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.createRequest(ClientInvoker.java:165) at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invokeSync(ClientInvoker.java:151) at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invoke(ClientInvoker.java:115) at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientProxy.invoke(ClientProxy.java:76) at com.sun.proxy.$Proxy33.getClusterInfo(Unknown Source) at io.zeebe.clustertestbench.handler.QueryClusterStateInCamundaCloudHandler.handle(QueryClusterStateInCamundaCloudHandler.java:29) at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.executeJob(JobRunnableFactory.java:44) at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.lambda$create$0(JobRunnableFactory.java:39) at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) at java.base/java.util.concurrent.FutureTask.run(Unknown Source) at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by korthout** [comment]
Update: Latest version is able to create the clusters and clients on its own without human interference. PR still requires some cleanup of commits, and then a review.

File: core/src/main/java/io/zeebe/clustertestbench/handler/CreateApiClientInCamundaCloudHandler.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      clientId = createZeebeClientResponse.getClientId();
+      LOGGER.info("Created client {} for cluster {}", clientId, clusterId);
+
+      ZeebeClientConnectiontInfo clientInfo;
+      var i = 0;
+      while ((clientInfo = getConnectionInfo(clusterId, clientId)) == null) {
+        // todo: remove this workaround after https://github.com/camunda-cloud/console/issues/2225
+        // After creating the cluster, the region information is not yet available in console.
+        // It takes some time before the info is updated.
+        Thread.sleep(Duration.ofSeconds(10).toMillis());
```
Comment (npepinpe) [inline core/src/main/java/io/zeebe/clustertestbench/handler/CreateApiClientInCamundaCloudHandler.java:54]: For such thing, I would look into https://jodah.net/failsafe/retry/. I've had a very good experience with failsafe - it's much lighter and easier to use than, say, Resilience4J imo (but less powerful).

File: core/src/main/java/io/zeebe/clustertestbench/handler/CreateApiClientInCamundaCloudHandler.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import io.zeebe.clustertestbench.testdriver.impl.CamundaCLoudAuthenticationDetailsImpl;
+import java.time.Duration;
+import java.util.regex.Pattern;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateApiClientInCamundaCloudHandler implements JobHandler {
+
+  // https://regex101.com/r/OVAMuf/1
```
Comment (npepinpe) [inline core/src/main/java/io/zeebe/clustertestbench/handler/CreateApiClientInCamundaCloudHandler.java:21]: :smile:

**Review by npepinpe — APPROVED** [review]
Looks good :+1:

**Comment by korthout** [comment]
bors merge

**Pull request merged**